apps/pie-docs/src/components/switch/code/code.md

```diff
@@ -0,0 +1,124 @@
+---
+eleventyNavigation:
+    key: Code
+    parent: Switch
+    order: 3
+shouldShowContents: true
+eleventyComputed:
+    props: "{% include './props.json' %}"
+---
+
+## Overview
+
+<p>
+  <a href="https://www.npmjs.com/@justeattakeaway/pie-switch">
+    <img alt="GitHub Workflow Status" src="https://img.shields.io/npm/v/@justeattakeaway/pie-switch.svg">
+  </a>
+</p>
+
+`pie-switch` is a Web Component built using the Lit library. It offers a simple and accessible switch component for web applications.
+
+This component can be easily integrated into various frontend frameworks and customized through a set of properties.
+
+## Installation
+
+To install `pie-switch` in your application, run the following on your command line:
+
+```shell
+# npm
+$ npm i @justeattakeaway/pie-switch
+```
+
+```shell
+# yarn
+$ yarn add @justeattakeaway/pie-switch
+```
+
+{% notification {
+  type: "neutral",
+  iconName: "link",
+  message: "For more information on using PIE components as part of an application, check out the [Getting Started Guide.](https://github.com/justeattakeaway/pie/wiki/Getting-started-with-PIE-Web-Components)."
+} %}
+
+## Playground
+
+ <iframe
+  src="https://webc.pie.design/?path=/story/switch--default&viewMode=story&shortcuts=true&singleStory=true"
+  width="100%"
+  height="600px"
+  style="border: none; margin-top: 32px;"
+></iframe>
+
+## Props
+
+{% componentDetailsTable {
+  tableData: props
+} %}
+
+## Events
+
+This component emits the `pie-switch-changed` event when switched. In order to add event listening to this component, you can treat it like a native HTML element in your application.
+
+For example, to add a click handler in various templates:
+
+## Examples
+
+For HTML:
+
+```js
+// import as module into a js file e.g. main.js
+import '@justeattakeaway/pie-switch'
+```
+
+```html
+<!-- pass js file into <script> tag -->
+<pie-switch id="switch" isChecked label="Label" labelPlacement="trailing"></pie-switch>
+<script>
+document.querySelector('#switch').addEventListener('pie-switch-changed', () => {
+    // Code to execute when switch is toggled
+});
+</script>
+```
+
+For Native JS Applications, Vue, Angular, Svelte etc.:
+
+```js
+// Vue templates (using Nuxt 3)
+import { PieSwitch } from '@justeattakeaway/pie-switch';
+
+<pie-switch isChecked label="Label" labelPlacement="trailing" @pie-switch-changed="handleChange"></pie-switch>
+```
+
+For React Applications.
+
+```js
+// React templates (using Next 13)
+import { PieSwitch } from '@justeattakeaway/pie-switch/dist/react';
+
+<pie-switch isChecked label="Label" labelPlacement="trailing" onSwitchChangedEvent={handleChange}></pie-switch>
```

Comment:
the event should be `onPieSwitchChanged`
```suggestion
<pie-switch isChecked label="Label" labelPlacement="trailing" onPieSwitchChanged={handleChange}></pie-switch>
```
<img width="1280" alt="Screenshot 2023-11-20 at 09 59 17" src="https://github.com/justeattakeaway/pie/assets/28605803/4d71cfc5-0674-4eeb-85fe-693571a5e175">